src/nwb_datajoint/decoding/sorted_spikes.py

```diff
@@ -0,0 +1,184 @@
+        return fetch_nwb(self, (AnalysisNwbfile, 'analysis_file_abs_path'), *attrs, **kwargs)
+
+    def fetch1_dataframe(self):
+        return self.fetch_dataframe()[0]
+
+    def fetch_dataframe(self):
+        return pd.concat([data['spike_indicator'].set_index('time') for data in self.fetch_nwb()], axis=1)
+
+
+def make_default_decoding_parameters_cpu():
+
+    classifier_parameters = dict(
+        environments=[_DEFAULT_ENVIRONMENT],
+        observation_models=None,
+        continuous_transition_types=_DEFAULT_CONTINUOUS_TRANSITIONS,
+        discrete_transition_type=DiagonalDiscrete(0.98),
+        initial_conditions_type=UniformInitialConditions(),
+        infer_track_interior=True,
+        knot_spacing=10,
+        spike_model_penalty=1E1
+    )
+
+    predict_parameters = {
+        'is_compute_acausal': True,
+        'use_gpu':  False,
+        'state_names':  ['Continuous', 'Uniform']
+    }
+    fit_parameters = dict()
+
+    return classifier_parameters, fit_parameters, predict_parameters
+
+
+def make_default_decoding_parameters_gpu():
+    classifier_parameters = dict(
+        environments=[_DEFAULT_ENVIRONMENT],
+        observation_models=None,
+        continuous_transition_types=_DEFAULT_CONTINUOUS_TRANSITIONS,
+        discrete_transition_type=DiagonalDiscrete(0.98),
+        initial_conditions_type=UniformInitialConditions(),
+        infer_track_interior=True,
+        knot_spacing=10,
+        spike_model_penalty=1E1
+    )
+
+    predict_parameters = {
+        'is_compute_acausal': True,
+        'use_gpu':  True,
+        'state_names':  ['Continuous', 'Fragmented']
+    }
+
+    fit_parameters = dict()
+
+    return classifier_parameters, fit_parameters, predict_parameters
+
+
+@schema
+class SortedSpikesClassifierParameters(dj.Manual):
+    definition = """
+    classifier_param_name : varchar(80) # a name for this set of parameters
+    ---
+    classifier_params :   BLOB    # initialization parameters
+    fit_params :          BLOB    # fit parameters
+    predict_params :      BLOB    # prediction parameters
+    """
+
+    def insert_default(self):
+        (classifier_parameters, fit_parameters,
+         predict_parameters) = make_default_decoding_parameters_cpu()
+        self.insert1(
+            {'classifier_param_name': 'default_decoding_cpu',
+             'classifier_params': classifier_parameters,
+             'fit_params': fit_parameters,
+             'predict_params': predict_parameters},
+            skip_duplicates=True)
+
+        (classifier_parameters, fit_parameters,
+         predict_parameters) = make_default_decoding_parameters_gpu()
+        self.insert1(
+            {'classifier_param_name': 'default_decoding_gpu',
+             'classifier_params': classifier_parameters,
+             'fit_params': fit_parameters,
+             'predict_params': predict_parameters},
+            skip_duplicates=True)
+
+    def insert1(self, key, **kwargs):
```

Comment:
It does not currently work for multiple entries. Well to be correct, the fetch will work but won't do the conversion to classes. I tried briefly but it didn't work so I took it out. I can try again but if it takes too long then I will just punt it to later because I think it's more important to get people decoding.

Reply:
Okay. It's a little complicated and I think most use cases involving fetching a single parameter set so I'd prefer to punt on this.